external-crates/move/crates/move-model/src/builder/model_builder.rs

```diff
@@ -27,26 +27,37 @@ use crate::{
 pub(crate) struct ModelBuilder<'env> {
     /// The global environment we are building.
     pub env: &'env mut GlobalEnv,
-    /// A symbol table for structs.
-    pub struct_table: BTreeMap<QualifiedSymbol, StructEntry>,
-    /// A reverse mapping from ModuleId/StructId pairs to QualifiedSymbol. This
+    /// A symbol table for datatypes.
+    pub datatype_table: BTreeMap<QualifiedSymbol, DatatypeEntry>,
+    /// A reverse mapping from ModuleId/DatatypeId pairs to QualifiedSymbol. This
     /// is used for visualization of types in error messages.
-    pub reverse_struct_table: BTreeMap<(ModuleId, DatatypeId), QualifiedSymbol>,
+    pub reverse_datatype_table: BTreeMap<(ModuleId, DatatypeId), QualifiedSymbol>,
     /// A symbol table for functions.
     pub fun_table: BTreeMap<QualifiedSymbol, FunEntry>,
     /// A symbol table for constants.
     pub const_table: BTreeMap<QualifiedSymbol, ConstEntry>,
 }
 
-/// A declaration of a struct.
+/// A declaration of a datatype.
 #[derive(Debug, Clone)]
-pub(crate) struct StructEntry {
+pub(crate) struct DatatypeEntry {
```

Comment:
Not this PR, but it would be nice to support positional datatypes at some point.